inference-engine/src/mkldnn_plugin/nodes/mkldnn_eltwise_node.cpp

```diff
@@ -1406,11 +1406,6 @@ void MKLDNNEltwiseNode::prepareParams() {
             bool canCollapse = true;
             for (int i = 0; i < dims_in.size(); i++) {
                 if (dims_in[i][dims_in[i].size() - 2] != 1) {
-                    if (dims_in[i][dims_in[i].size() - 1] == 1) {
```

Comment:
For me it looks like this condition was valid. Eltwise kernel should distinguish between two collapse dims cases: {16, 1} -> {16} and {1, 16} -> {16} because its not clear do we need to broadcast value or make a simple vector load. I think your tests cases pass because you have identical shapes on first and second inputs. Please check eltwise with inputs: 0: {16, 16, 16, 16, 16}, 1: {16, 16, 16, 16, 1}.

Reply:
We discussed offline. It's ok because on 1397 line we have check for case with different last dims. It covers inputs [1.8.5.5] and [1.8.5.1].
TODO: should add tests for this case